src/main/java/com/google/devtools/build/lib/analysis/ConfiguredRuleClassProvider.java

```diff
@@ -813,13 +814,18 @@ public BuildOptions createBuildOptions(OptionsClassProvider optionsProvider) {
       ImmutableMap<String, Object> skylarkAccessibleTopLevels,
       ImmutableList<Bootstrap> bootstraps,
       ImmutableList<Class<?>> modules) {
-    ImmutableMap.Builder<String, Object> envBuilder = ImmutableMap.builder();
+    ImmutableMap.Builder<Identifier, Object> envBuilder = ImmutableMap.builder();
 
     SkylarkModules.addSkylarkGlobalsToBuilder(envBuilder);
     for (Class<?> module : modules) {
       Runtime.setupModuleGlobals(envBuilder, module);
     }
-    envBuilder.putAll(skylarkAccessibleTopLevels.entrySet());
+    envBuilder.putAll(
+        skylarkAccessibleTopLevels
+            .entrySet()
+            .stream()
+            .collect(
+                ImmutableMap.toImmutableMap(e -> Identifier.of(e.getKey()), e -> e.getValue())));
```

Comment:
An Identifer is a syntax node.  Only the parser should be in the business of creating them.